spec/views/devise/two_factor_authentication/show.html.slim_spec.rb

```diff
@@ -3,58 +3,36 @@
 describe 'devise/two_factor_authentication/show.html.slim' do
   context 'user has unconfirmed_mobile and is two_factor_enabled' do
     it 'only mentions mobile' do
-      allow(view).to receive(:current_user).
-        and_return(create(:user, :tfa_confirmed, unconfirmed_mobile: '5005550006'))
+      user = create(:user, :signed_up, unconfirmed_mobile: '5005550999')
+      allow(view).to receive(:current_user).and_return(user)
+      @user_decorator = UserDecorator.new(user)
 
       render
 
-      expect(rendered).to have_content 'A one-time passcode has been sent to 5005550006.'
+      expect(rendered).to have_content 'A one-time passcode has been sent to +1 (500) 555-0999.'
     end
   end
 
-  context 'user has 2 second factors but no unconfirmed email' do
-    it 'mentions both email and mobile' do
-      allow(view).to receive(:current_user).
-        and_return(
-          create(
-            :user, :signed_up, :both_tfa_confirmed, email: 'foo@bar.com', mobile: '5005550006'
-          )
-        )
-
-      render
-
-      expect(rendered).
-        to have_content 'A one-time passcode has been sent to foo@bar.com and +1 (500) 555-0006.'
-    end
-  end
-
-  context 'user only has email second factor and no unconfirmed email' do
-    it 'mentions email' do
-      allow(view).to receive(:current_user).
-        and_return(create(:user, :signed_up, email: 'foo@bar.com'))
+  context 'user has second factor and no unconfirmed email' do
+    it 'mentions mobile' do
+      user = create(:user, :signed_up, :with_mobile, mobile: '5005550777')
```

Comment:
This can be simplified to `user = create(:user, :signed_up)`, and then in the expectation below, we can check `"A one-time passcode has been sent to #{user.mobile}"`